proto/substrait/algebra.proto

```diff
@@ -52,6 +53,22 @@ message RelCommon {
       substrait.extensions.AdvancedExtension advanced_extension = 10;
     }
   }
+
+  // Optional metadata that the producer may attach to the relation.
+  // Consumer should hold no expectations about the existance metadata.
+  message Metadata {
+    // Name (alias) for this relation.
+    // If provided, must be unique across the relations in the plan/root.
+    // Safe to ignore, or can be used for e.g. qualifying the relation (see e.g.
+    // Spark's SubqueryAlias), or debugging.
+    // Repeated to allow for multiple levels of naming, e.g. catalog/schema/table.
+    repeated string alias = 1;
```

Comment:
Having this separate is the only part left of the PR that could be contentious.  By allowing this to qualify the relation this is participating in the evaluation of the plan.  Naming a relation isn't going to be enough to fix Datafusion's duplicate field errors -- the equivalent of SELECT x+2, x+2 will still trigger them.